http-filter-example/http_filter.cc

```diff
@@ -23,55 +23,57 @@ HttpSampleDecoderFilterConfig::HttpSampleDecoderFilterConfig(
     : config_(config) {
-  // populate list of valid operations, implementation will be changed in the next few commits
-  operations_.insert("set-header");
   
   const std::string header_config = headerValue();
 
-  // find the position of the first space character
-  const size_t spacePos = header_config.find(' ');
-  if (spacePos == std::string::npos) {
-    fail("no arguments provided");
-    setError(1);
-    return;
-  }
+  // split by operation (comma delimited config)
+  auto operations = StringUtil::splitToken(header_config, "\n", false, true);
 
-  // can't be a string view
-  const auto operation = header_config.substr(0, spacePos);
+  // process each operation
+  for (auto const& operation : operations) {
+    auto tokens = StringUtil::splitToken(operation, " ");
+    if (tokens.size() < 3) {
+      fail("too few arguments provided");
+      setError(1); // TODO: set error based on globally defined macros
+      return;
+    }
 
-  // error checking -- invalid operation
-  if (operations_.find(operation) == operations_.end()) {
-    fail("invalid operation provided");
-    setError(1);
-    return;
-  }
+    // determine if it's request/response
+    if (tokens[0] != "http-request" && tokens[0] != "http-response") {
+      fail("first argument must be <http-response/http-request>");
+      setError(1); // TODO: set error based on globally defined macros
+      return;
+    }
+    bool isRequest = (tokens[0] == "http-request");
 
-  const std::string args = header_config.substr(spacePos + 1);
+    // TODO: determine the operation type (right now I'm assuming that it's always a set-header operation)
 
-  // create a string stream to iterate over the arguments
-  std::istringstream iss(args);
-  std::vector<std::string> values;
-  std::string value;
+    // make new header processor
+    SetHeaderProcessor* processor = new SetHeaderProcessor();
 
-  // split the rest of the string by space and store the values in a vector
-  while (iss >> value) {
-      values.push_back(value);
-  }
+    // parse operation
+    if(processor->parseOperation(tokens)) {
+      fail("unable to parse operation");
+      setError(1); // TODO: set error based on globally defined macros
+      return;
+    }
 
-  // error checking -- wrong number of arguments
-  if (operation == "set-header" && values.size() != 2) {
-    fail("expected 2 arguments");
-    setError(1);
-    return;
+    // keep track of operations to be executed
+    if (isRequest) {
+      request_header_processors_.push_back(processor);
+    }
   }
-
-  // insert the key-value pair into header_ops
-  header_ops_[operation] = values;
 }
 
-HttpSampleDecoderFilter::~HttpSampleDecoderFilter() {}
+HttpSampleDecoderFilter::~HttpSampleDecoderFilter() {
+  onDestroy(); // not sure if this is called automatically somewhere else
+}
 
-void HttpSampleDecoderFilter::onDestroy() {}
+void HttpSampleDecoderFilter::onDestroy() {
+  for (auto processor : request_header_processors_) {
+    free(processor);
```

Comment:
This is a bug: the processor memory was initialised with `new` above: You have to match `new`/`delete`, `malloc`/`free`, and `new[]` with `delete[]`. Mixing them is a bug (technically Undefined Behaviour).
But, you can avoid the risk altogether with `shared_ptr` or `unique_ptr`.